api-server-nest/src/announcements/announcements.service.ts

```diff
@@ -22,7 +22,12 @@ export class AnnouncementsService {
     ensureOne(announcement: CreateOrUpdateAnnouncementRequest): Promise<AnnouncementModel> {
         const whereAnyAnnouncementExists = sequelize.literal('true = true')
 
-        return this.announcementModel.findOrCreate({ where: whereAnyAnnouncementExists, defaults: announcement })
+        const id = uuidv4()
+
```

Comment:
^ got to provide a possible id for the case where the announcement doesn't exist